providers/base/bin/camera_test.py

```diff
@@ -505,30 +524,47 @@ def _capture_image_gstreamer(self, filename, width, height, pixelformat):
             pipeline.add(bayer2rgb)
             caps.link(bayer2rgb)
             rgb_capture = bayer2rgb
-
         else:
+            mime_type = "video/x-raw"
+            if pixelformat == "MJPG":
+                mime_type = "image/jpeg"
             caps.set_property(
                 "caps",
                 self.Gst.Caps.from_string(
-                    "video/x-raw,width={},height={}".format(width, height)
+                    "{},width={},height={}".format(mime_type, width, height)
                 ),
             )
             pipeline.add(caps)
             rgb_capture = caps
 
-        # Add encoder
-        encoder = self.Gst.ElementFactory.make("jpegenc", "encoder")
-        pipeline.add(encoder)
+        # Add valve, note that we can't rely on the default value of drop
+        # it varies by installations
+        valve = self.Gst.ElementFactory.make("valve", "photo-valve")
+        assert valve, "Valve element could not be created"
```

Comment:
Assertions should be used mainly for debugging purposes and could be turned off by some compilation options.
It's better practice to raise an exception:
```suggestion
        if not valve:
            raise RuntimeError("Valve element could not be created")
```